app/src/main/java/com/twilio/video/app/ui/room/RoomViewState.kt

```diff
@@ -11,13 +13,18 @@ data class RoomViewState(
     val participantThumbnails: List<ParticipantViewState>? = null,
     val selectedDevice: AudioDevice? = null,
     val availableAudioDevices: List<AudioDevice>? = null,
-    val isLobbyLayoutVisible: Boolean = true,
-    val isConnectingLayoutVisible: Boolean = false,
-    val isConnectedLayoutVisible: Boolean = false,
+    val layoutState: LayoutState = Lobby,
     val isCameraEnabled: Boolean = false,
     val localVideoTrack: VideoTrackViewState? = null,
     val isMicEnabled: Boolean = false,
     val isAudioMuted: Boolean = true,
     val isVideoOff: Boolean = false,
-    val isScreenCaptureOn: Boolean = false
+    val isScreenCaptureOn: Boolean = false,
+    val roomStats: RoomStats? = null
 ) : UIState()
+
+sealed class LayoutState {
```

Comment:
What's the benefit of a sealed class versus an enum here?

Reply:
I tend to prefer Sealed classes over Enum classes in Kotlin. They provide more flexibility as you can provide additional properties if needed for subclasses and some cool auto complete IDE capabilities such as auto complete for all conditions when writing when statements.